csharp/sources/Valkey.Glide/Internals/Request.SetCommands.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+using Valkey.Glide.Commands.Constants;
+
+using static Valkey.Glide.Internals.FFI;
+
+namespace Valkey.Glide.Internals;
+
+internal partial class Request
+{
+    public static Cmd<long, bool> SetAddAsync(ValkeyKey key, ValkeyValue value)
+    {
+        GlideString[] args = [key.ToGlideString(), value.ToGlideString()];
+        return new(RequestType.SAdd, args, false, response => response == 1);
+    }
+
+    public static Cmd<long, long> SetAddAsync(ValkeyKey key, ValkeyValue[] values)
+    {
+        GlideString[] args = [key.ToGlideString(), .. values.ToGlideStrings()];
+        return Simple<long>(RequestType.SAdd, args);
+    }
+
+    public static Cmd<long, bool> SetRemoveAsync(ValkeyKey key, ValkeyValue value)
+    {
+        GlideString[] args = [key.ToGlideString(), value.ToString()];
+        return new(RequestType.SRem, args, false, response => response == 1);
+    }
+
+    public static Cmd<long, long> SetRemoveAsync(ValkeyKey key, ValkeyValue[] values)
+    {
+        GlideString[] args = [key.ToGlideString(), .. values.Select((v) => v.ToString())];
+        return Simple<long>(RequestType.SRem, args);
+    }
+
+    public static Cmd<HashSet<object>, ValkeyValue[]> SetMembersAsync(ValkeyKey key)
+    {
+        GlideString[] args = [key.ToGlideString()];
+        return new(RequestType.SMembers, args, false, set => set.Select(obj => (ValkeyValue)obj.ToString()).ToArray());
```

Comment:
avoid `ToString()`
```suggestion
        return new(RequestType.SMembers, args, false, set => set.Cast<ValkeyValue>().Select(vv => (GlideString)vv).ToArray());
```
or if double cast works
```suggestion
        return new(RequestType.SMembers, args, false, set => set.Cast<ValkeyValue>().Cast<GlideString>().ToArray());
```
(probably one cast to GS is enough)

Reply:
Is this backwards? We want to go from `HashSet<object>` to `ValkeyValue[]` but wouldn't `set.Cast<ValkeyValue>().Select(vv => (GlideString)vv).ToArray());` first try to cast the object to a `ValkeyValue`, and then create a `GlideString[]`?

Reply:
oh yes